### DISCUSSION TITLE
Busy week 13-17 June

### DISCUSSION THREAD

**Discussion by paoloserra** [caracal-pipeline/caracal#1411] [General]
# Tasks

A first list of possible items to work on at the busy week. Feel free to add comments/thoughts/explanations/new items.

- Caracal-2/Stimela-2 for a selection of workers (Sphe, Oleg)
- DDcal (Francesca after Pol imager, maybe in Caracal-2)
- ~Polcal?~ Pol imager (Francesca)
- SoFiA-2 and #1320 (Dane, Paolo, Athanaseus)
- u=0 flagging task (Filippo, Paolo, Erwin)
- MosaicSteward -> MosaicQueen (Athanaseus, Paolo)
- Parallelize MosaicQueen (Sphe, Paolo)
- Resize masks also spectrally, see below (Julia, Erwin)
- Include imcontsub in Caracal (Josh, Sphe, Filippo) REMARK (Josh): this can be done in one day
- Mosaicing -> include equolver to convolve cubes to the same resolution, at least make a start (Josh)
- Crystalball #1222 (most work had been done! Paolo)
- 1GC with Quartical in Caracal-2 (Landman, Jon)
- Separate crosscal'd and selfcal'd MS files to save disk space (all)

# Attendees

If your name is missing from this list please send an email to Paolo.

Names in parenthesis are from people who would like to come but cannot confirm yet.

- Paolo
- Oleg
- Erwin
- Dane
- Filippo
- Lexy
- Francesca
- Athanaseus
- Julia
- Josh
- Landman

It will be possible to join parts of the telecon remotely. People who've signed up for this are:

- Kshitij
- Jon
- Sphe 

# Compute requirements

We think we only need to connect to our remote machines and transfer small files during the busy week, so we shouldn't need to be too demanding. If you have thoughts on this please add them here.

**Comment by Fil8** [comment]
- [ ] resize masks also spectrally 

   - usefult for cubes with narrower spectral ranges than the defaults
   - useful for using the same masks over binned/unbinned cubes

**Comment by paoloserra** [comment]
@SpheMakh image-based continuum subtraction is already in the list above. My idea was to include in CARACal Josh's script available at https://github.com/gigjozsa/jscripts . However, the DCT method you've just showed is also very interesting! Maybe we can do both.

**Comment by paoloserra** [comment]
For continuum subtraction see also https://github.com/caracal-pipeline/caracal/issues/141 .

**Comment by paoloserra** [comment]
Wrap-up

- Josh: imcontsub implementation in line worker nearly completed, needs testing.
- Lexy: included tons of packages in Stimela and took care of some dodgy CASA errors.
- Dane: prevent repeated cont model subtraction/addition; SoFiA-2 in Stimela (PR open), plan for SoFiA-2 implementation into CARACal with new parameters in place.
- Francesca: pol image worker ready, it makes full Stokes cubes, images, primary beams. Testing is ongoing (Francesca + Kshitij) before PR.
- Filippo: u=0 stripe flagging implemented, only thing left to do is to do the FFT in astropy instead of casatools  to not add new dependencies.
- Paolo: finalised MosaicQueen v1.1.1 (now parallelised), helped with testing of u=0 stripe flagging.
- Erwin, optimizing SoFiA-2 parameters, test mask regridding
- Landman: Stimela-2 crosscal worker using Quartical; image convolution / spectral index tools.
- Julia: implemented adapting the user clean mask to the user cube along the spectral axis: includes going from wide to narrow chans, from narrow to wide chans, and making a channel selection. Testing ongoing, PR to be opened.
- Oleg: Stimela-2 development.
- Sphe: release of MosaicQueen (done) + include it in Stimela (ongoing)
- Kshitij: testing pol image worker.
